src/request.rs

```diff
@@ -386,13 +388,32 @@ impl<State> Request<State> {
         Ok(res)
     }
 
-    /// returns a `Cookie` by name of the cookie.
+    /// Returns a `Cookie` by name of the cookie.
     #[must_use]
     pub fn cookie(&self, name: &str) -> Option<Cookie<'static>> {
         self.ext::<CookieData>()
             .and_then(|cookie_data| cookie_data.content.read().unwrap().get(name).cloned())
     }
 
+    /// Returns a signed `Cookie` by name of the cookie if it verified successfully.
+    #[cfg(feature = "secure-cookies")]
+    #[cfg_attr(feature = "docs", doc(cfg(feature = "secure-cookies")))]
+    #[must_use]
+    pub fn signed_cookie(&self, key: &Key, name: &str) -> Option<Cookie<'static>> {
```

Comment:
This seems awkward, passing the key everywhere? I don't think that's the right way to do this, more details in review comment.